cmake/sdlchecks.cmake

```diff
@@ -608,6 +608,15 @@ macro(CheckWayland)
         if(PKG_LIBDECOR_FOUND)
             set(HAVE_WAYLAND_LIBDECOR TRUE)
             set(HAVE_LIBDECOR_H 1)
+            cmake_push_check_state()
+            set(CMAKE_CHECK_REQUIRED_FLAGS "${CMAKE_CHECK_REQUIRED_FLAGS} ${PKG_LIBDECOR_CFLAGS}")
+            set(CMAKE_CHECK_REQUIRED_LIBRARIES "${CMAKE_CHECK_REQUIRED_FLAGS} ${PKG_LIBDECOR_LDFLAGS}")
```

Comment:
Oops
```suggestion
            set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} ${PKG_LIBDECOR_CFLAGS}")
            set(CMAKE_REQUIRED_LIBRARIES "${CMAKE_REQUIRED_FLAGS} ${PKG_LIBDECOR_LDFLAGS}")
```

Reply:
I caught that too.  I also changed the first line to `cmake_push_check_state(RESET)` to blank the CMake required variables, so no need to concatenate strings.

Also moved to using `check_symbol_exists()`, which is already used elsewhere and accomplishes the same thing, so there is no need to pull in another module just for this check.  Everything should be supported back to CMake 3.0, so it should be good to backport to SDL2.

Reply:
Sounds good to me.
I refrained myself from adding `RESET` to other `cmake_push_check_state`'s  because of the following code in SDL2:
https://github.com/libsdl-org/SDL/blob/8efa1f8fc60e0a2961ce3bad9293c048301d25ba/CMakeLists.txt#L294-L298

But when looking at [the Emscripten CMake toolchain file](https://github.com/mosra/toolchains/blob/master/generic/Emscripten.cmake), I can't see any settings of `CMAKE_REQUIRED_*` (nor in its history).

Reply:
If there is an off chance that clearing them could cause problems in some cases (maybe cross compiling?), then might as well put it back since not clearing it doesn't break anything.

Reply:
I doubt the original comment, because when cross compiling you should have configured `CMAKE_<LANG>_FLAGS` correctly anyways such that the compiler gets the correct flags in the non-`CMAKE_REQUIRED_*` case.
The documentation of [`CHeckCSouceCompiles`](https://cmake.org/cmake/help/latest/module/CheckCSourceCompiles.html) says that `CMAKE_C_FLAGS` is added anyways, but this is not shown on the other documentation pages.